assets/js/components/Table/Table.test.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { screen, fireEvent } from '@testing-library/react';
+import 'intersection-observer';
+import '@testing-library/jest-dom';
+
+import { faker } from '@faker-js/faker';
+import { Factory } from 'fishery';
+
+import { renderWithRouter } from '@lib/test-utils';
+
+import { filterTable } from '@lib/test-utils/table';
+
+import Table from './Table';
+
+const tableConfig = {
+  pagination: true,
+  columns: [
+    {
+      title: 'Column1',
+      key: 'column1',
+      filter: true,
+      filterFromParams: true,
+    },
+    {
+      title: 'Column2',
+      key: 'column2',
+      filter: true,
+      filterFromParams: true,
+    },
+    {
+      title: 'Column3',
+      key: 'column3',
+      filter: true,
+      filterFromParams: true,
+    },
+  ],
+};
+
+const tableFactory = Factory.define(() => ({
+  column1: faker.name.firstName(),
+  column2: faker.name.firstName(),
+  column3: faker.name.firstName(),
+}));
+
+function TestTable({ config, data }) {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  return (
+    <Table
+      config={config}
+      data={data}
+      searchParams={searchParams}
+      setSearchParams={setSearchParams}
+    />
+  );
+}
+
+describe('Table component', () => {
+  describe('filtering', () => {
+    it('should filter by the chosen filter option', () => {
+      const data = tableFactory.buildList(10);
+      const { column1: value1 } = data[0];
+
+      renderWithRouter(<TestTable config={tableConfig} data={data} />);
+
+      filterTable('Column1', value1);
+
+      const table = screen.getByRole('table');
+      expect(table.querySelectorAll('tbody > tr')).toHaveLength(1);
```

Comment:
You don't need to `renderWithRouter` the `TestTable` here, you could just render the `Table` with empty search params and a noop as `setSearchParams`